app/hooks/useContracts.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+
+export function useContracts({ network }: { network: string }) {
+  const [
+    { xMetricJson, questionAPIJson, questionStateController, bountyQuestionJson, costController, vaultJson },
+    setContracts,
+  ] = useState(getContracts(network));
+
+  useEffect(() => {
+    setContracts(getContracts(network));
+  }, [network]);
+
+  return {
+    xMetricJson,
+    questionAPIJson,
+    questionStateController,
+    bountyQuestionJson,
+    costController,
+    vaultJson,
+  };
+}
+
+function getContracts(network: string) {
+  let xMetricJson;
+  let questionAPIJson;
+  let questionStateController;
+  let bountyQuestionJson;
+  let costController;
+  let vaultJson;
+
+  if (network === "polygon" || !network) {
```

Comment:
If no network it should fail or default to localhost not polygon

Reply:
Also lets move this to a server side call in the api folder with routes and rename file to server but once it's in there like meta.ts you may not need to rename, so it doesn't get bundled client side like it does currently, even in the previous implementation I had I guess it was being bundled client side even though it was only being called in the useLoader server side call

Reply:
Also we are loading all these JSON files into the client bundle when all we need is the abi and address and it loads it several times.  The loading of contracts only needs to be done once but here on page load it's loaded several times
if you log out it in here you'll see in console
<img width="1570" alt="Screen Shot 2022-09-19 at 1 41 13 PM" src="https://user-images.githubusercontent.com/1916496/191094319-cfc4e618-686f-454b-9f0e-48a9064a8e2c.png">

Reply:
Is it confirmed that the useLoader payload when correctly implemented was being bundled client side? The most recent commit places all contract interaction in its route's loader, but I can move forward with putting it in a standalone api route if so. Not super familiar with remix so this has been a learning experience. Any guidance here would be much appreciated

Reply:
I was under the impression useLoader acted as a "server" execution environment?

Either way - @jdaly13 let's get this PR in and we open tickets for this one if it is a problem.  I know Bryan and Seb are gonna bring some of their architecture experience from the flipside apps over as well this week